ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/accordion/v1/accordion/clientlibs/contentframe/js/accordion.js

```diff
@@ -13,92 +13,98 @@
+         *
+         * @private
+         */
+        function onDocumentReady() {
 
-        var elements = document.querySelectorAll(Accordion.selectors.self);
-        for (var i = 0; i < elements.length; i++) {
-            new Accordion({ element: elements[i] });
-        }
+            var elements = document.querySelectorAll(Accordion.selectors.self);
+            for (var i = 0; i < elements.length; i++) {
+                new Accordion({element: elements[i]});
+            }
 
-        var MutationObserver = window.MutationObserver || window.WebKitMutationObserver || window.MozMutationObserver;
-        var body = document.querySelector("body");
-        var observer = new MutationObserver(function(mutations) {
-            mutations.forEach(function(mutation) {
-                // needed for IE
-                var nodesArray = [].slice.call(mutation.addedNodes);
-                if (nodesArray.length > 0) {
-                    nodesArray.forEach(function(addedNode) {
-                        if (addedNode.querySelectorAll) {
-                            var elementsArray = [].slice.call(addedNode.querySelectorAll(Accordion.selectors.self));
-                            elementsArray.forEach(function(element) {
-                                new Accordion({ element: element });
-                            });
-                        }
-                    });
-                }
+            var MutationObserver = window.MutationObserver || window.WebKitMutationObserver || window.MozMutationObserver;
+            var body = document.querySelector("body");
+            var observer = new MutationObserver(function (mutations) {
+                mutations.forEach(function (mutation) {
+                    // needed for IE
+                    var nodesArray = [].slice.call(mutation.addedNodes);
+                    if (nodesArray.length > 0) {
+                        nodesArray.forEach(function (addedNode) {
+                            if (addedNode.querySelectorAll) {
+                                var elementsArray = [].slice.call(addedNode.querySelectorAll(Accordion.selectors.self));
+                                elementsArray.forEach(function (element) {
+                                    new Accordion({element: element});
+                                });
+                            }
+                        });
+                    }
+                });
             });
-        });
 
-        observer.observe(body, {
-            subtree: true,
-            childList: true,
-            characterData: true
-        });
+            observer.observe(body, {
+                subtree: true,
+                childList: true,
+                characterData: true
+            });
+        }
+
+        if (document.readyState !== "loading") {
+            onDocumentReady();
+        } else {
+            document.addEventListener("DOMContentLoaded", onDocumentReady);
```

Comment:
Can there be any negative case here, can there be case `DOMContentLoaded` is not registered yet but event is already dispatched. As in first iteration of loop only `window.addEventListener('AccordionMixinLoaded', onAccordionMixinLoaded)` seems to be registered.

Reply:
This should work, please ignore .